### DISCUSSION TITLE
[Dev Tools] Foundry deploy errors but HH works

### DISCUSSION THREAD

**Discussion by 0xBeans** [zkSync-Community-Hub/zksync-developers#748] [Dev Tooling] [answered]
### Team or Project

Abstract

### Environment

Testnet

### Select the Dev Tool you are using

Foundry

### Provide the version of the tool (if applicable)

forge 0.0.2 (0b24fcc 2024-10-24T00:36:14.606654000Z)

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

<img width="1256" alt="Screenshot 2024-10-28 at 5 30 25 PM" src="https://github.com/user-attachments/assets/514516e4-2e8d-4d33-ac64-0c29d0cde868">

Have you guys ever seen this before?

I deployed with HH just fine using a simple deploy.ts, but would really like to deploy with Foundry as the deploy script complicated.

### Repo Link (Optional)

https://github.com/voteagora/ERC20VotesPartialDelegationUpgradeable

**Comment by alhenaai** [comment]
It seems you are encountering issues deploying with Foundry on the ZKsync network, despite successful deployments using Hardhat. This could be due to differences in how Foundry handles deployments compared to Hardhat, especially in terms of transaction handling and contract interactions on ZKsync.

Here are a few steps you can take to troubleshoot and potentially resolve the issue:

1. **Check Foundry Configuration**: Ensure that your Foundry setup is correctly configured for ZKsync. This includes the correct RPC URLs and any specific settings required by ZKsync.

2. **Version Compatibility**: Verify that the version of Foundry you are using (`forge 0.0.2`) is compatible with the current ZKsync testnet configurations. Sometimes, updates in the testnet might require updates in the tools.

3. **Error Logs**: Analyze the error shown in the screenshot you provided. Unfortunately, I can't view the screenshot directly, but errors typically provide clues on what might be going wrong.

4. **Simplify the Deployment Script**: If the Foundry script is complicated, try simplifying it to match the Hardhat script's functionality as closely as possible to test if the issue persists with a simpler contract or setup.

5. **Consult Documentation and Community**: Review the [Foundry deployment guide](https://docs.zksync.io/build/tooling/foundry/migration-guide/deployment) for any specific instructions or known issues related to ZKsync. You can also seek further assistance and share your experiences on the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

If these steps do not resolve the issue, it might be beneficial to share more specific error messages or logs for further analysis.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by Karrq** [comment, marked as answer]
`foundry-zksync` [recently](https://github.com/matter-labs/foundry-zksync/pull/636) had an update to address this issue, does the issue persist if you update your locally installed version?
